src/Utilities/StdHelpers.hpp

```diff
@@ -389,3 +391,16 @@ inline std::array<T, Dim> operator-(const std::array<T, Dim>& rhs) noexcept(
   }
   return result;
 }
+
+template <size_t Dim, typename T>
+inline T magnitude(const std::array<T, Dim>& rhs) noexcept(
+    noexcept(T{}) and
+    noexcept(std::declval<std::array<T, Dim>&>() += rhs) and
+    noexcept(square(rhs[0]) and
+             noexcept(std::sqrt(rhs[0])))) {
```

Comment:
As far as I can tell, `noexcept(std::sqrt(...))` is always false, but `std::sqrt` never throws exceptions, so we should decide what we want to do about that.

Also, the parentheses are nested incorrectly here, as indicated by the indenting.